src/fastmcp/server/server.py

```
@@ -954,10 +882,23 @@ async def get_tools(self) -> dict[str, Tool]:
+            pass
+
+        # Try providers
+        for provider in self._providers:
+            try:
+                tool = await provider.get_tool(key)
+                if tool is not None:
+                    return tool
+            except NotFoundError:
+                continue
+
+        raise NotFoundError(f"Unknown tool: {key}")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)